exporters/failure/collector_azure_devops.py

```diff
@@ -0,0 +1,174 @@
+        metadata=env_vars("AZURE_DEVOPS_PRIORITY"),
+    )
+
+    def __attrs_post_init__(self):
+        credentials = BasicAuthentication("", self.token)
+        connection = Connection(base_url=self.tracker_api.url, creds=credentials)
+        self.client = connection.clients_v6_0.get_work_item_tracking_client()
+
+    def get_work_items(self) -> List[WorkItem]:
+        logging.info("Getting work items")
+
+        try:
+            wiql = Wiql(query="Select [System.Id] From WorkItems")
+            wiql_results = self.client.query_by_wiql(wiql).work_items
+            return [
+                self.client.get_work_item(int(result.id)) for result in wiql_results
+            ]
+        except AzureDevOpsServiceError as error:
+            logging.error(FailureProviderAuthenticationError.auth_message)
+            raise FailureProviderAuthenticationError from error
+        except Exception as error:
+            # TODO test limitation 200?!?!
+            logging.error(error)  # pragma: no cover
+            raise  # pragma: no cover
+
+    def filter_by_type(self, work_item_type: str) -> bool:
+        if not self.work_item_type:
+            return True
+        return work_item_type in self.work_item_type
+
+    def filter_by_priority(self, priority: str) -> bool:
+        if not self.work_item_priority:
+            return True
+        return str(priority) in self.work_item_priority
+
+    def filter_by_project(self, project: str) -> bool:
+        if not self.projects:
+            return True
+        return project in self.projects
+
+    def get_app_name(self, work_item: WorkItem) -> str:
+        try:
+            labels: str = work_item.fields["System.Tags"]
+            labels = labels.split("; ")
+
+            label_text = self.app_label + "="
+
+            for label in labels:
+                if label_text in label:
+                    return label.replace(label_text, "")
+            return "unknown"
+        except KeyError:
+            return "unknown"
+
+    def search_issues(self) -> list[TrackerIssue]:
+        """
+        To maintain consistency, we call this method `search_issues`. An
+        `issue` in Azure DevOps is called `work item`.
+        """
+        production_work_items = []
+        for work_item in self.get_work_items():
+            is_production_bug = (
+                self.filter_by_type(work_item.fields["System.WorkItemType"])
+                and self.filter_by_priority(
+                    work_item.fields["Microsoft.VSTS.Common.Priority"]
+                )
+                and self.filter_by_project(work_item.fields["System.TeamProject"])
+            )
+
+            if is_production_bug:
+                created_at = work_item.fields["System.CreatedDate"]
+                work_item_id = work_item.id
+                title = work_item.fields["System.Title"]
+
+                created_tz = parse_assuming_utc(created_at, _DATETIME_FORMAT)
```

Comment:
Are we sure azure will always have same date format or do we want to make it configurable parameter?

Reply:
from my tests, it was always the same

Reply:
Ok so it's specified here:
  https://learn.microsoft.com/en-us/rest/api/storageservices/formatting-datetime-values

While date is always there, the time may be, however for that work item the time probably always will be present.